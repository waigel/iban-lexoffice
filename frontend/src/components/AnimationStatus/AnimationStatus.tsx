import React, {useEffect} from 'react';
import lottie from 'lottie-web';

import lottie_error_status from '../../lottie/18053-no-error-cancelled.json';
import lottie_success_status from '../../lottie/114089-add-to-cart-checkmark-animation.json';

export interface AnimationStatusProps {
  status: 'success' | 'failed';
}
export const AnimationStatus = ({status}: AnimationStatusProps) => {
  const lottieId =
    status === 'success' ? 'lottie_success_status' : 'lottie_error_status';
  useEffect(() => {
    lottie.destroy(lottieId);
    lottie.loadAnimation({
      container: document.querySelector(`#${lottieId}`) as Element,
      animationData:
        status === 'success' ? lottie_success_status : lottie_error_status,
      name: lottieId,
      loop: false,
    });
  }, []);

  return <div id={lottieId} />;
};
