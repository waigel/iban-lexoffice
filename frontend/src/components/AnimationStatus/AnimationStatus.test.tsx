import React from 'react';
import TestRenderer from 'react-test-renderer';
import {AnimationStatus} from './AnimationStatus';
import 'jest-canvas-mock';

it('should test the AnimationStatus component rendering', () => {
  let tree = TestRenderer.create(<AnimationStatus status="failed" />);
  expect(tree.toJSON()).toMatchSnapshot();
  tree = TestRenderer.create(<AnimationStatus status="success" />);
  expect(tree.toJSON()).toMatchSnapshot();
});
