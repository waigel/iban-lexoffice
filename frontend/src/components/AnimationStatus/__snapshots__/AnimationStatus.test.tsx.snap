// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the AnimationStatus component rendering 1`] = `
<div
  id="lottie_error_status"
/>
`;

exports[`should test the AnimationStatus component rendering 2`] = `
<div
  id="lottie_success_status"
/>
`;
